### PR TITLE
Provide basic Dockerfile

### PR DESCRIPTION
**Pull request by balboah**
Example with mvn build through docker.
This allows users to easily start experimenting with no other dependencies than Docker.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,10 @@
+FROM maven:3.5-jdk-8
+
+ARG MAVEN_OPTS="-Xmx1024M -Xss128M -XX:MetaspaceSize=512M -XX:MaxMetaspaceSize=1024M -XX:+CMSClassUnloadingEnabled"
+COPY . /usr/src/app
+WORKDIR /usr/src/app
+
+RUN mvn clean package -DskipTests
+RUN chmod +x /usr/src/app/bin/pkg/*.sh
+
+ENTRYPOINT [ "./uReplicator-Distribution/target/uReplicator-Distribution-pkg/bin/start-controller.sh" ]
```
Comment (fx19880617) [inline Dockerfile]: Shall we have the same docker image for both controller and server?
Better to have a new start-ureplicator.sh and you can specify role[controller/server] there.

**Comment by balboah** [comment]
@fx19880617 see updated commit.

Example of how it can be used now:
```
docker run -ti <image name> controller <controller params>
docker run -ti <image name> worker <worker params>
docker run -ti <image name> bash
```

File: Dockerfile [hunk 2]
```diff
@@ -7,4 +7,7 @@ WORKDIR /usr/src/app
 RUN mvn clean package -DskipTests
 RUN chmod +x /usr/src/app/bin/pkg/*.sh
 
-ENTRYPOINT [ "./uReplicator-Distribution/target/uReplicator-Distribution-pkg/bin/start-controller.sh" ]
+COPY docker-entrypoint.sh /docker-entrypoint.sh
+
+ENTRYPOINT [ "/docker-entrypoint.sh" ]
+CMD [ "controller" ]
```
Comment (fx19880617) [inline Dockerfile]: I guess we don't need this .

Comment (balboah) [reply Dockerfile]: only if you don't want a default :)

**Comment by balboah** [comment]
Not that I need this merged, just wanted to push what I had to do locally. But you should setup some schedule for checking (and merging) PR's more often ;) I don't have a green button to press!

**Comment by xhl1988** [comment]
Sorry for the delay. Thanks for contributing to uReplicator :)

**Comment by daluu** [comment]
It would probably be nice to add mention of docker support & how to run the docker image in the README. I actually missed this until I was searching something in the issues list and came across this, granted I should have looked at the files in the root of the repo.

Also, looking at this Dockerfile & my initial issue building for docker, makes me curious whether the maven build process heap memory constraints is only limited to Docker, because I notice this Dockerfile sets specific memory requirements, and I hit the memory error compiling uReplicator-Controller, following the link it referenced mentioned to adjust the memory settings for maven, and that fixed my issue. Wonder if it would ever happen locally on a non-Docker maven build. If possible for non-Docker, might want to mention it. If only for docker, might want to mention it for docker.

**Pull request merged**